### PR TITLE
Add drone pipeline builds for docker+linux+macos

### PR DESCRIPTION
**Pull request by hjpotter92**

### CODE REVIEW

File: .drone.star [hunk 1]
```diff
@@ -0,0 +1,136 @@
+PLATFORMS = [
+    {"os": "linux", "arch": "amd64"},
+    {"os": "linux", "arch": "arm64"},
+    {"os": "darwin", "arch": "amd64"},
```
Comment (iameli) [inline .drone.star:4]: Anything blocking `{darwin: "arm64"}` on the other Mac Mini?

File: .drone.star [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+def get_docker_tags(repo, prefix, branch, commit):
+    tags = [
+        "latest",
+        "catalyst",
+        branch.replace("/", "-"),
+        "$DRONE_BUILD_CREATED",
+        commit,
+        commit[:8],
+    ]
+    return ["%s:%s-%s" % (repo, prefix, tag) for tag in tags]
```
Comment (iameli) [inline .drone.star:25]: I'm not sure I like `latest` getting pushed for every build; while technically true I wouldn't except `docker run livepeer/catalyst` to give me whatever build on a branch has gone out first.

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+	&& make -j$(nproc) install
+
+ENV	LD_LIBRARY_PATH="/src/compiled/lib" \
+	C_INCLUDE_PATH="/src/compiled/include"
+
+COPY	.	.
+
+RUN	mkdir -p build/ \
+	&& cd build/ \
+	&& cmake .. -DPERPETUAL=1 -DLOAD_BALANCE=1 -DCMAKE_INSTALL_PREFIX=/opt -DCMAKE_PREFIX_PATH=/src/compiled -DCMAKE_BUILD_TYPE=RelWithDebInfo \
```
Comment (iameli) [inline Dockerfile:36]: Let's add the flag to make shared binaries with `libmist.a` here.

(And getting each flag on its own line wouldn't be a terrible idea)

**Review by iameli — APPROVED** [review]
Nice. Added some feedback.

File: .drone.star (hunk 1 above)
Comment (hjpotter92) [reply .drone.star:4]: nothing except that i had not yet provided drone-runner with updated config.

File: .drone.star (hunk 2 above)
Comment (hjpotter92) [reply .drone.star:25]: yes. i wanted to experiment with the build context before updating this. the tags generated would (should?) be:
- branch_name_in_snake_case
- commit sha
- short commit sha
- latest only if branch-name == catalyst
- some CI unique identifier

File: .drone.star (hunk 1 above)
Comment (hjpotter92) [reply .drone.star:4]: darwin-arm64 is not supported by drone runners :(

File: Dockerfile (hunk 3 above)
Comment (hjpotter92) [reply Dockerfile]: added this in recent updates to dockerfile

**Pull request merged**